### ISSUE TITLE
ConnectionError(e, request=request)

### ISSUE THREAD

**Issue by Unloaded123456** [arek-kwapis/rustedanvil#7]
I have downloaded "requests" with pip, but there is still a problem:

`requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=25566): Max retries exceeded with url: /Minecraft.Download/indexes/legacy.json (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000002144092F190>: Failed to establish a new connection: [WinError 10061] '))`

**Comment by arek-kwapis** [comment]
The script you are trying to run isn't meant to be run without prior setup. You need a XAMPP server configured in a way so that SSL is port 25566, and you need to place the "legacy.json" file on the server with the proper folder structure.

The error is more or less telling you no server exists on localhost:25566, and thus it can't find the file required. 

Ran without prior setup of the workspace, this is the correct outcome.

**Comment by Unloaded123456** [comment]
1- What is XAMPP? Do I need to install manually?
2- How to set up? I could not find any script for that kind of setup.
3- What IP address I should set? 0.0.0.0, 255.255.255.255, 128.128.128.128 or what?

**Comment by arek-kwapis** [comment]
1. XAMPP is a developmental webserver toolkit that we're temporarily using to essentially emulate a webserver
2. There is no script and there won't be one. In Milestone 2, we will stop using XAMPP in favour of a native implementation (a nodejs local private webserver that will be made during setup, and then destroyed once setup is finished, making it more secure and less risky for users)
3. Setup is as easy as running the provided XAMPP installer provided in the redist folder and allocating SSL of the Apache webserver to port 25566. Do note that with XAMPP there are **major** security risks even if used properly. 
3.1. Setting it up for the download script right now is kind of pointless - we use a custom legacy.json file only available on my local machine currently. Once we reach Milestone 2 and the need for it arises, It'll be uploaded to the Abstrat CDN for the scripts to use

**Comment by Unloaded123456** [comment]
What version of XAMPP that I need? The latest, oldest or somewhere between it?

**Comment by arek-kwapis** [comment]
It's provided in the redist folder

https://github.com/Abstrat-Technologies/rustedanvil/tree/1.6.4_965/_REDIST

**Comment by Unloaded123456** [comment]
Thanks for assistance

**Comment by Unloaded123456** [comment]
Where should I put the legacy.json file? Does the built-in one work?

**Comment by arek-kwapis** [comment]
This is already covered in my comment from a few days ago

> 3.1. Setting it up for the download script right now is kind of pointless - we use a custom legacy.json file only available on my local machine currently. Once we reach Milestone 2 and the need for it arises, It'll be uploaded to the Abstrat CDN for the scripts to use
